### PR TITLE
feat: sdk action publishing improvements

### PR DESCRIPTION
**Pull request by ryan-timothy-albert**
**Existing Tag Issue**
Improves this [issue](https://linear.app/speakeasy/issue/SPE-3438/publish-action-run-even-though-sdk-version-was-not-bumped)

Sometimes not so infrequently in examples like this [PR](https://github.com/dubinc/dub-node/commit/9bd96c83d3415df3cf1acc977c6e5804d678b346#diff-4a00447b47c36ebd3ded77321deff288e424f4605496ee1199fc9953a20f4473) the SDK version stays the same but the speakeasy version changes in Releases.MD. Then when create_release runs it fails because the tag already exists, this confuses our users. We can check if we are failing because a tag already exists and just move forward with no publishing for safety but without a miscellaneous unclear failure.

Another way we could solve this problem is
- For package managers where we know you can replace version we delete the release/tag, create a new release/tag, move forward with publishing. This seems like it does have some risks to me though. I'm wondering if just not moving forward with publishing is better when the tag already exists.

**Pre Release Version Issue**
We don't support prerelease versions right now in sdk-generation-action. You're able to place a prerelease version in gen.yaml and generation goes fine but our action wasn't actually creating releases for them. I fixed this problem by changing our regex for parsing releases. [[TESTED HERE](https://github.com/ryan-timothy-albert/sample-go-sdk/releases/tag/v2.1.8-alpha.2)]

### CODE REVIEW

File: .github/workflows/sdk-publish.yaml [hunk 1]
```diff
@@ -93,7 +93,7 @@ jobs:
       - name: Tune GitHub-hosted runner network
         uses: smorimoto/tune-github-hosted-runner-network@v1
       - id: release
-        uses: speakeasy-api/sdk-generation-action@v15
+        uses: speakeasy-api/sdk-generation-action@v15.11.0
```
Comment (ryan-timothy-albert) [inline .github/workflows/sdk-publish.yaml:96]: will remove

File: action.yml [hunk 2]
```diff
@@ -147,7 +147,7 @@ outputs:
     description: "The directory the SDK target was generated to"
 runs:
   using: "docker"
-  image: "docker://ghcr.io/speakeasy-api/sdk-generation-action:v15"
+  image: "docker://ghcr.io/speakeasy-api/sdk-generation-action:v15.11.0"
```
Comment (ryan-timothy-albert) [inline action.yml:150]: ditto

File: internal/git/releases.go [hunk 3]
```diff
@@ -61,14 +61,25 @@ func (g *Git) CreateRelease(releaseInfo releases.ReleasesInfo) error {
...
 				Name:            github.String(fmt.Sprintf("%s - %s - %s", lang, tag, environment.GetInvokeTime().Format("2006-01-02 15:04:05"))),
 				Body:            github.String(fmt.Sprintf(`# Generated by Speakeasy CLI%s`, releaseInfo)),
 			})
 			if err != nil {
-				return fmt.Errorf("failed to create release: %w", err)
+				if release, _, err := g.client.Repositories.GetReleaseByTag(context.Background(), os.Getenv("GITHUB_REPOSITORY_OWNER"), getRepo(), *tagName); err == nil && release != nil {
+					fmt.Println(fmt.Sprintf("a github release with tag %s already existing ... skipping publishing", *tagName))
+					fmt.Println(fmt.Sprintf("to publish this version again delete the github tag and release"))
+					if _, ok := outputs[fmt.Sprintf("publish_%s", lang)]; ok {
+						outputs[fmt.Sprintf("publish_%s", lang)] = "false"
```
Comment (chase-crumbaugh) [inline internal/git/releases.go:76]: I realize we do this in other places but I don't think its ideal to take in a loosely typed map and then infer things about its contents, like that it has keys in the format `publish_<lang>`. There are many ways to improve that--one option would be to add a helper like `func shouldPublishLanguage(lang string, outputs map[string]string)`

**Review by chase-crumbaugh — APPROVED** [review]
One small thing

File: internal/git/releases.go (hunk 3 above)
Comment (ryan-timothy-albert) [reply internal/git/releases.go:76]: Wouldn't the function just do the same thing. Check if it exists if it does set it to false? In this case

**Comment by ryan-timothy-albert** [comment]
@chase-crumbaugh and I discussed. We will allow publishing to move forward if the tag already is there. In the future we may want to integrate directly with package managers to also see if this version had been published before that package manager

**Pull request merged**